C++17 fix

leviathanch opened the pull request:
C++ 17 and upwards requires std::allocator to be replaced with std::allocator_traits, in order to compile successfully.

Commit:
C++17 fix

C++ 17 and upwards requires std::allocator to be replaced with
std::allocator_traits, in order to compile successfully.

limbo018 commented on limbo/thirdparty/lemon/lemon/bits/array_map.h:
This fix may not be backward compatible to versions earlier than C++11. Please add version guard to ensure backward compatibility.

Commit:
C++17 fix

C++ 17 and upwards requires std::allocator to be replaced with
std::allocator_traits, in order to compile successfully.

leviathanch commented:
I applied a guard as requested and did a git push -f
